src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -183,12 +199,55 @@ public function addContent($content, $type = null)
      */
     public function addHtmlContent($content, $charset = 'UTF-8')
     {
-        $internalErrors = libxml_use_internal_errors(true);
-        $disableEntities = libxml_disable_entity_loader(true);
+        $dom = $this->useHtml5Parser ? $this->parseHtml5($content, $charset) : $this->parseXhtml($content, $charset);
+        $this->addDocument($dom);
 
-        $dom = new \DOMDocument('1.0', $charset);
-        $dom->validateOnParse = true;
+        $base = $this->filterRelativeXPath('descendant-or-self::base')->extract(array('href'));
+
+        $baseHref = current($base);
+        if (\count($base) && !empty($baseHref)) {
+            if ($this->baseHref) {
+                $linkNode = $dom->createElement('a');
+                $linkNode->setAttribute('href', $baseHref);
+                $link = new Link($linkNode, $this->baseHref);
+                $this->baseHref = $link->getUri();
+            } else {
+                $this->baseHref = $baseHref;
+            }
+        }
+    }
+
+    /**
+     * @param string $content The HTML content
+     * @param string $charset The charset
+     *
+     * @return \DOMDocument
+     */
+    private function parseHtml5($content, $charset = 'UTF-8')
+    {
+        set_error_handler(function () { throw new \Exception(); });
+
+        // Convert to UTF-8
+        try {
+            $content = mb_convert_encoding($content, 'UTF-8', $charset);
+        } catch (\Exception $e) {
+            // Handle invalid charsets by fallbacking on ISO-8859-2
```

Comment:
why ISO-8859-2? the second most used charset is WINDOWS-1252, I think that'd be a better choice